client/network/src/service.rs

```diff
@@ -1019,7 +1028,7 @@ impl<B: BlockT + 'static, H: ExHashT> Future for NetworkWorker<B, H> {
 			// Process the next message coming from the `NetworkService`.
 			let msg = match this.from_worker.poll_next_unpin(cx) {
 				Poll::Ready(Some(msg)) => msg,
-				Poll::Ready(None) => return Poll::Ready(Ok(())),
+				Poll::Ready(None) => return Poll::Ready(()),
```

Comment:
This `return Poll::Ready(())` is used to signal not to poll the `NetworkWorker` again.

> You **must** call this method in a loop.

With this comment from above in mind one would still continue polling the `NetworkWorker`, right? Am I missing something?